File: src/main/java/be/hubau/sane/parser/Displayer.java
package be.hubau.sane.parser;/* -----------------------------------------------------------------------------
 * Displayer.java
 * -----------------------------------------------------------------------------
 *
 * Producer : com.parse2.aparse.Parser 2.5
 * Produced : Mon Apr 15 15:26:09 CEST 2019
 *
 * -----------------------------------------------------------------------------
 */

import java.util.ArrayList;

public class Displayer implements Visitor {

    public Object visit(Rule_sane rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_expression rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_ws rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_newline rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_comment_start_symbol rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_non_eol rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_comment rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_keyval rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_key rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_simple_key rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_unquoted_key rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_quoted_key rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_dotted_key rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_dot_sep rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_keyval_sep rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_val rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_string rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_basic_string rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_quotation_mark rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_basic_char rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_basic_unescaped rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_escaped rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_escape rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_escape_seq_char rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_ml_basic_string rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_ml_basic_string_delim rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_ml_basic_body rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_ml_basic_char rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_ml_basic_unescaped rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_literal_string rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_apostrophe rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_literal_char rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_ml_literal_string rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_ml_literal_string_delim rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_ml_literal_body rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_ml_literal_char rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_integer rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_minus rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_plus rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_underscore rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_digit1_9 rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_digit0_7 rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_digit0_1 rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_hex_prefix rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_oct_prefix rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_bin_prefix rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_dec_int rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_unsigned_dec_int rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_hex_int rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_oct_int rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_bin_int rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_float rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_float_int_part rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_frac rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_decimal_point rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_zero_prefixable_int rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_exp rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_special_float rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_inf rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_nan rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_boolean rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_true rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_false rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_list rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_list_open rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_list_close rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_list_values rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_list_sep rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_ws_comment_newline rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_map rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_map_open rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_map_close rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_map_sep rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_map_pair rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_map_keyvals rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_ALPHA rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_DIGIT rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Rule_HEXDIG rule) {
        return visitRules(rule.rules);
    }

    public Object visit(Terminal_StringValue value) {
        System.out.print(value.spelling);
        return null;
    }

    public Object visit(Terminal_NumericValue value) {
        System.out.print(value.spelling);
        return null;
    }

    private Object visitRules(ArrayList<Rule> rules) {
        for (Rule rule : rules)
            rule.accept(this);
        return null;
    }
}

/* -----------------------------------------------------------------------------
 * eof
 * -----------------------------------------------------------------------------
 */

File: src/main/java/be/hubau/sane/parser/Displayer.kt
package be.hubau.sane.parser

import java.util.*

class Displayer : Visitor {

    override fun visit(rule: Rule_newline): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_ws): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_expression): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_sane): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_comment_start_symbol): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_non_eol): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_comment): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_keyval): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_key): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_simple_key): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_unquoted_key): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_quoted_key): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_dotted_key): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_dot_sep): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_keyval_sep): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_val): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_string): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_basic_string): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_quotation_mark): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_basic_char): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_basic_unescaped): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_escaped): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_escape): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_escape_seq_char): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_ml_basic_string): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_ml_basic_string_delim): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_ml_basic_body): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_ml_basic_char): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_ml_basic_unescaped): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_literal_string): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_apostrophe): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_literal_char): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_ml_literal_string): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_ml_literal_string_delim): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_ml_literal_body): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_ml_literal_char): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_integer): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_minus): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_plus): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_underscore): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_digit1_9): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_digit0_7): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_digit0_1): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_hex_prefix): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_oct_prefix): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_bin_prefix): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_dec_int): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_unsigned_dec_int): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_hex_int): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_oct_int): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_bin_int): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_float): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_float_int_part): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_frac): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_decimal_point): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_zero_prefixable_int): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_exp): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_special_float): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_inf): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_nan): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_boolean): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_true): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_false): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_list): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_list_open): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_list_close): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_list_values): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_list_sep): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_ws_comment_newline): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_map): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_map_open): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_map_close): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_map_sep): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_map_pair): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_map_keyvals): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_ALPHA): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_DIGIT): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(rule: Rule_HEXDIG): Any? {
        return visitRules(rule.rules)
    }

    override fun visit(value: TerminalStringValue): Any? {
        print(value.spelling)
        return null
    }

    override fun visit(value: TerminalNumericValue): Any? {
        print(value.spelling)
        return null
    }

    private fun visitRules(rules: ArrayList<Rule>?): Any? {
        if (rules != null) {
            for (rule in rules)
                rule.accept(this)
        }
        return null
    }
}